rules/scripts/run_test.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+#
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script will run the command line arguments as-is
+# and will further save the output to a file if the env
+# variable OT_RUN_TEST_OUR_DIR is specified. In this case
+# the output log will be $OT_RUN_TEST_OUR_DIR/<path>.log
+# where <path> is derived from the bazel test target
+# and test run number.
+#
+# An example usage of this script is as follows:
+#
+# bazel test \
+#   --test_env=OT_RUN_TEST_OUR_DIR=/tmp/log_directory/ \
+#   --run_under=//rules/scripts:run_test \
+#   //sw/device/silicon_creator/rom/e2e/...
+#
+
+if [ -z "$OT_RUN_TEST_OUR_DIR" ]; then
+  echo "OT_RUN_TEST_OUR_DIR not provided, will not log output"
+  "$@"
+else
+  # variables are documented here:
+  # https://bazel.build/reference/test-encyclopedia
+  if [ -z "$TEST_RUN_NUMBER" ]; then
+    RUN_NR=""
+  else
+    RUN_NR=".$TEST_RUN_NUMBER"
+  fi
+  OUT_FILE="$OT_RUN_TEST_OUR_DIR/$TEST_TARGET$RUN_NR.log"
+  mkdir -p "$(dirname $OUT_FILE)"
```

Comment:
yes, this should be fixed. In order to use `set -e`, some minor restructing will need to take place because we want to execute some code even after a test has been executed and failed.